src/classes/modules/java.base/java/text/DateFormat.java

```diff
@@ -0,0 +1,83 @@
+package java.text;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+import java.util.TimeZone;
+
+public abstract class DateFormat extends Format {
+
+    static int nInstances;
+    private int id = nInstances++;
+    public static final int FULL = 0;
+    public static final int LONG = 1;
+    public static final int MEDIUM = 2;
+    public static final int SHORT = 3;
+    public static final int DEFAULT = SHORT;
+    protected Calendar calendar;
+
+    public DateFormat() {}
+
+    public static final DateFormat getDateTimeInstance(int dateStyle, int timeStyle, Locale locale) {
+        DateFormat dateFormat = getDT(dateStyle, timeStyle, 3, locale);
+        return dateFormat;
+    }
+
+    private static DateFormat getDT(int dateStyle, int timeStyle, int flags, Locale locale) {
```

Comment:
Isn't "flags" perhaps a misnomer here? If it refers to the static constants above, it would be "style", as flags can be combined independently of each other, but the style can only be one out of these four. (If this is the case, I would rename the parameter to "style" even if OpenJDK uses the wrong terminology.)

Reply:
Yes Cyrille, thanks for pointing this out. Parameter name now changed to "style"